src/react/components/common/assetPreview/videoAsset.test.tsx

```diff
@@ -107,124 +127,133 @@ describe("Video Asset Component", () => {
+            wrapper.find(CustomVideoPlayerButton).at(3).simulate("click");
 
-    it("moves to the previous tagged frame when clicking the back button", () => {
-        const childAssets = MockFactory.createChildVideoAssets(defaultProps.asset);
-        const currentAsset = childAssets[4];
-        const expectedAsset = {
-            ...childAssets[3],
-            state: AssetState.NotVisited,
-        };
+            expect(videoPlayerMock.prototype.pause).toBeCalled();
+            expect(videoPlayerMock.prototype.seek).toBeCalledWith(expectedAsset.timestamp);
+            expect(onChildSelectedHandler).toBeCalledWith(expectedAsset);
+        });
 
-        videoPlayerMock.prototype.getState = jest.fn(() => {
-            return {
-                player: {
-                    currentTime: currentAsset.timestamp,
-                },
-            };
+        it("blocks moving to next tagged from when navigation is cancelled", () => {
+            const currentAsset = childAssets[0];
+
+            onBeforeAssetChangedHandler.mockImplementationOnce(() => false);
+
+            setupMoveFrameTest(currentAsset);
+            wrapper.find(CustomVideoPlayerButton).at(3).simulate("click");
+
+            expect(videoPlayerMock.prototype.pause).not.toBeCalled();
+            expect(videoPlayerMock.prototype.seek).not.toBeCalled();
+            expect(onChildSelectedHandler).not.toBeCalled();
         });
 
-        const props: IVideoAssetProps = {
-            ...defaultProps,
-            childAssets,
-            timestamp: currentAsset.timestamp,
-        };
+        it("moves to the previous tagged frame when clicking the back button", () => {
+            const currentAsset = childAssets[4];
+            const expectedAsset = {
+                ...childAssets[3],
+                state: AssetState.NotVisited,
+            };
 
-        wrapper = createComponent(props);
-        mockLoaded();
+            setupMoveFrameTest(currentAsset);
 
-        wrapper.find(CustomVideoPlayerButton).at(2).simulate("click");
+            wrapper.find(CustomVideoPlayerButton).at(3).simulate("click");
+            wrapper.find(CustomVideoPlayerButton).at(2).simulate("click");
 
-        expect(videoPlayerMock.prototype.pause).toBeCalled();
-        expect(videoPlayerMock.prototype.seek).toBeCalledWith(expectedAsset.timestamp);
-        expect(onChildSelectedHandler).toBeCalledWith(expectedAsset);
-    });
+            expect(videoPlayerMock.prototype.pause).toBeCalled();
+            expect(videoPlayerMock.prototype.seek).toBeCalledWith(expectedAsset.timestamp);
+            expect(onChildSelectedHandler).toBeCalledWith(expectedAsset);
+        });
 
-    it("moves to the next expected frame when clicking the next button", () => {
-        const childAssets = MockFactory.createChildVideoAssets(defaultProps.asset);
-        const currentAsset = childAssets[0];
+        it("blockings moving to the previous tagged frame when navigation is cancelled", () => {
```

Comment:
Nit: I would put the navigation cancelled tests inside another describe

Reply:
They are all about navigating to and from assets - so I will keep them here for now.